build-tools/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -759,7 +764,8 @@ public void user(Map<String, String> userSpec) {
     }
 
     private File getBuildPluginFile(String name) {
-        return project.getRootProject().file("build-tools/src/main/resources/" + name);
+        URL resource = getClass().getResource(name);
+        return fileOperations.getResources().getText().fromUri(resource).asFile();
```

Comment:
Nice! I forgot we can use `TextResources` to load stuff off the classpath. We should probably replace [stuff like this](https://github.com/elastic/elasticsearch/blob/master/build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/info/GlobalBuildInfoPlugin.java#L358-L374) as well.